extensions/amp-ad-network-adsense-impl/0.1/adsense-a4a-config.js

```diff
@@ -64,5 +88,31 @@ export function adsenseIsA4AEnabled(win, element, useRemoteHtml) {
     addExperimentIdToElement(experimentId, element);
     forceExperimentBranch(win, ADSENSE_A4A_EXPERIMENT_NAME, experimentId);
   }
+
+  // If not in the unconditioned canonical experiment, attempt to
+  // select into the undiluted canonical experiment.
+  const inUnconditionedCanonicalExp = !!getExperimentBranch(
+      win, ADSENSE_EXP_NAMES.UNCONDITIONED_CANONICAL);
+  if (!inUnconditionedCanonicalExp && isCanonical(win)) {
+    selectAndSetExperiments(win, element,
+        [ADSENSE_EXPERIMENTS.CANONICAL_EXP,
+          ADSENSE_EXPERIMENTS.CANONICAL_CTL],
+        ADSENSE_EXP_NAMES.CANONICAL);
+  }
+}
+
+/**
+ * @param {!Window} win
+ * @param {!Element} element
+ * @param {boolean} useRemoteHtml
+ * @returns {boolean}
+ */
+export function adsenseIsA4AEnabled(win, element, useRemoteHtml) {
+  randomlySelectUnconditionedExperiments(win, element);
+  if (useRemoteHtml || !supportsNativeCrypto(win) ||
```

Comment:
Couple of things:
- if you leave the native crypto check then non-ssl canonical pages will not be allowed to use Fast Fetch
- I don't understand why you're not checking the experiment here?  Seems to me that you're enabling FF for ALL (currently SSL) canonical traffic which is not what you want.